pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -51,52 +52,69 @@ func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
 
 	for idx := range rs.PodSpec.Containers {
 		if rs.PodSpec.Containers[idx].Name == "" {
-			rs.PodSpec.Containers[idx].Name = cfg.Defaults.UserContainerName(ctx)
-		}
-
-		if rs.PodSpec.Containers[idx].Resources.Requests == nil {
-			rs.PodSpec.Containers[idx].Resources.Requests = corev1.ResourceList{}
-		}
-		if _, ok := rs.PodSpec.Containers[idx].Resources.Requests[corev1.ResourceCPU]; !ok {
-			if rsrc := cfg.Defaults.RevisionCPURequest; rsrc != nil {
-				rs.PodSpec.Containers[idx].Resources.Requests[corev1.ResourceCPU] = *rsrc
-			}
-		}
-		if _, ok := rs.PodSpec.Containers[idx].Resources.Requests[corev1.ResourceMemory]; !ok {
-			if rsrc := cfg.Defaults.RevisionMemoryRequest; rsrc != nil {
-				rs.PodSpec.Containers[idx].Resources.Requests[corev1.ResourceMemory] = *rsrc
+			if len(rs.PodSpec.Containers) > 1 {
+				rs.PodSpec.Containers[idx].Name = cfg.Defaults.UserContainerName(ctx) + "-" + strconv.Itoa(idx)
```

Comment:
So strictly theoretically I might name my _other_ sidecars in a way that will cause a collision here. Do we care about that?

Reply:
Yep In order to have unique name added this condition check

Reply:
which check? This one won't deal with the case when I name my sidecar
`cfg.Defaults.UserContainerName(ctx) + "-" + 1` and don't name my main user container.

Reply:
It deals with that i mean
service.yaml file with name given for sidecars and no name given for main container

```
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  name: multicontainer
spec:
  template:
    metadata:
    spec:
      containers:
      - image: savita3020/container-a:v2
        ports:
        - containerPort: 8111
      - image: savita3020/container-b:v2
        name: second
      - image: savita3020/container-c:v2
        name: third
```
Output
```
spec:
  containerConcurrency: 0
  containers:
  - image: savita3020/container-a:v2
    name: user-container-0
    ports:
    - containerPort: 8111
    readinessProbe:
      successThreshold: 1
      tcpSocket:
        port: 0
    resources: {}
  - image: savita3020/container-b:v2
    name: second
    resources: {}
  - image: savita3020/container-c:v2
    name: third
    resources: {}
  timeoutSeconds: 300
status:
  conditions:
```

as main container name not given it uses a generated name

Please let me know if this is the question you had or something i misunderstood

Reply:
so what happenss if I do
```
...
      containers:
      - image: savita3020/container-a:v2
        ports:
        - containerPort: 8111
      - image: savita3020/container-b:v2
        name: user-container-0
      - image: savita3020/container-c:v2
        name: third
```

Reply:
In this case it will fail because of unique name
```
failed to create deployment "multicontainer-w9mtz-deployment": Deployment.apps "multicontainer-w9mtz-deployment" is invalid: spec.template.spec.containers[1].name: Duplicate value: "user-container-0"
```
Thank you for pointing it out
I will handle this part 
Instead of using `strconv.Itoa(idx)` i will generate unique name same like how k8s generates podname